screens/AllUsersListScreen/index.js

```diff
@@ -11,21 +12,35 @@ import RenderImagePairs from './RenderImagePairs';
 import {getDataUsersFromFile} from '../../utils/ApiUtils';
 import {theme} from '../../assets/theme/theme';
 import {GlobalDataContext} from '../../contexts/context';
+import Warning from '../../components/Warning/Warning';
 import styles from './style';
 
-const UsersScreen = ({navigation}) => {
+const AllUsersListScreen = ({navigation}) => {
   const {imageUrl} = useContext(GlobalDataContext);
-  const [data, setData] = useState([]);
+  const [initialData, setInitialData] = useState([]);
+  const [data, setData] = useState(initialData);
   const [searchItemValue, setSearchItemValue] = useState('');
 
   useEffect(() => {
     const fetchData = () => {
       const result = getDataUsersFromFile();
-      setData(result);
+      setInitialData(result);
     };
     fetchData();
   }, []);
 
+  const handleSearch = () => {
+    const result = initialData.filter(item => {
+      let param = item.fullName.toLowerCase();
+      return param.indexOf(searchItemValue) > -1;
+    });
+    setData(result);
+  };
+
+  useEffect(() => {
+    return handleSearch();
+  }, [initialData, searchItemValue, setData]);
```

Comment:
it's not a good practice to use outside function in useEffect, you should define that function inside useEffect

Reply:
isn't there eslint warning for this?

Reply:
No, eslint didn't warn. but I removed the function from depends.